### ISSUE TITLE
[ENH] Implement a Merger to create a univariate time series out of a Panel

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#5350]
**Is your feature request related to a problem? Please describe.**
If you have n forecasts or time series segments that may overlap and you want to get one single time series out of this, you need to merge all values with the same temporal information. 

This is required to enable #4924 since the cINN is creating time series segments of the length 24 for each time step. To get a point forecast, these overlapping samples must be merged. 

**Describe the solution you'd like**
The merger should temporally align all segments in the Panel and then merge them by applying different strategies. Required strategies are calculating 
* the mean 
* the median
over all values with the same temporal information

**Describe alternatives you've considered**
If this is to specific, this could also be integrated directly into the cINN forecaster (#4924)

**Comment by fkiraly** [comment]
I wonder, is this the first example of "panel-to-single-series" in sktime?

**Comment by fkiraly** [comment]
idea: would it not be nice if we could build the following type of pipeline, also perhaps with @astrogilda's tsBootstrap library:

`MyBootstrap() * my_forecaster * my_aggregator`, where `my_aggregator` might be any panel-to-series transformer, and the framework automatically upbroadcasts `my_forecaster` to the panel?

**Comment by benHeid** [comment]
That sounds cool. This would require a stride parameter for the merger currently implemented in the PR.

* stride = 0 -> Useful for the bootstrap example
* stride = 1 -> Needed by the cINN-based Forecaster
* stride > 1 -> Not sure, but perhaps useful for some cases.

**Comment by fkiraly** [comment]
> This would require a stride parameter for the merger currently implemented in the PR.

Can you explain what that would do?

**Comment by benHeid** [comment]
> > This would require a stride parameter for the merger currently implemented in the PR.
> 
> Can you explain what that would do?

With stride I mean something like how many one step shifts between the different instances are made. Perhaps there is a better term for it. I use stride since I have the feeling this is something similar as the stride parameter in convolutional neural networks.

Probably the best to explain it with an example:
* Consider that time steps are on x axis and n_instances on y axis.
* n_instances = 4
* horizon = 3 (or time steps)

### Example Stride 0
```
xxx
xxx
xxx
xxx
```

### Example Stride 1
```
xxx...
.xxx..
..xxx.
...xxx
```

### Example Stride 2
```
xxx......
..xxx....
....xxx..
......xxx
```

If I am doing the calculations correctly, the resulting time series length should then be: horizon + stride * (n_instances - 1)

**Comment by benHeid** [comment]
> what is pre-transform, what is post-transform, in the picture above?

I am unsure what you are referring to. In the picture above, the resulting time series after calculating the mean/median is the output of the merger. 
Does this answer the question? If not, it would probably be easier if we discuss it on Friday in the dev meeting.

**Comment by fkiraly** [comment]
> Does this answer the question? If not, it would probably be easier if we discuss it on Friday in the dev meeting.

Yes, let's do that. For quick summary, I don't know what is input, output, index, in relation to the merger transformer operation - I think a direct discussion is quickest to resolve my confusion.

**Comment by benHeid** [comment]
Preparation for dev meeting, to explain what happens:

```python
y = np.array([[[1, 2]], [[3,4]], [[5,6]], [[7,8]]])
result1 = Merger(method="mean", stride=0).fit_transform(y)
print(result1)
# 4, 5 [PS, the shape is (2,1,1), not sure which tags I need to set, to enable shape (2,1) )
# Internally here over all instances separtely a median is calculated
# 1,2
# 3, 4
# 5, 6
# 7, 8
# ----- mean
#4, 5 


result2 = Merger(method="mean", stride=1).fit_transform(y)
print(result2)
# 1, 2.5, 4.5, 6.5, 8 
# Internally the n_instances are shifted by one, and then the columnwise median is calculated
# 1, 2, -, - ,-, -
# -, 3, 4, -, -
# -, -, 5, 6, -
# -, -, -, 7, 8
# ------------ mean
# 1, 2.5, 4.5, 6.5, 8


result3 = Merger(method="mean", stride=2).fit_transform(y)
print(result3)
# 1, 2, 3, 4, 5, 6, 7, 8
# Internally the n_instances are shifted by two, and then the columnwise median is calculated
# 1, 2, -, - ,-, -
# -, -,  3, 4, -, -
# -, -,  -, -, 5, 6, -
# -, -, -, -, -, -, 7, 8
# ------------ mean
# 1, 2, 3, 4, 5, 6, 7, 8
```



Regarding the index: 
* I think, the best solution would be to use the time steps and consider the shifts.
